src/Tools/Metrics/Program.cs

```diff
@@ -316,7 +316,7 @@ async Task computeProjectMetricDataAsync(MSBuildWorkspace workspace, string proj
 
                 cancellation.ThrowIfCancellationRequested();
                 var compilation = await project.GetCompilationAsync(CancellationToken.None).ConfigureAwait(false);
-                var metricData = await CodeAnalysisMetricData.ComputeAsync(compilation.Assembly, compilation, CancellationToken.None).ConfigureAwait(false);
+                var metricData = await CodeAnalysisMetricData.ComputeAsync(compilation.Assembly, new CodeMetricsAnalysisContext(compilation, CancellationToken.None)).ConfigureAwait(false);
```

Comment:
Yep, we need to update this one in a follow-up PR. We can probably just link in all the files that are needed for parsing editorconfig options into this project, and create a dummy diagnostic descriptor for CA1501 to pass in the correct func here.

Reply:
See #3275